app/services/backup_code_generator.rb

```diff
@@ -24,6 +24,7 @@ def create
 
   # @return [Boolean]
   def verify(plaintext_code)
+    return false unless plaintext_code
```

Comment:
check for empty strings too?
```suggestion
    return false unless plaintext_code.present?
```